Route53 failed to specify region

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

> terraform -v
> Terraform v0.12.2

### Affected Resource(s)

resourceAwsRoute53DelegationSet(),
resourceAwsRoute53QueryLog(),
resourceAwsRoute53Record(),
resourceAwsRoute53ZoneAssociation(),
resourceAwsRoute53Zone(),
resourceAwsRoute53HealthCheck(),

### Terraform Configuration Files

```
# configure AWS provider for China region. 
provider "aws" {
  region 	= "cn-northwest-1"
  access_key	= AK
  secret_key	= SK
  
  endpoints {
    route53	= "https://api.route53.cn"
  }
}

# create a route 53 pub zone
resource "aws_route53_zone" "test" {
  
  name = "bobtest3.cn"
  comment = "try to create a pub zone with terraform in China region v2!"
}
```

### Debug Output

> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/15 12:15:49 [DEBUG] Creating Route53 hosted zone: {
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4:   CallerReference: "terraform-20190615041549099100000001",
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4:   HostedZoneConfig: {
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4:     Comment: "try to create a pub zone with terraform in China region"
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4:   },
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4:   Name: "bobtest2.cn"
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: }
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/15 12:15:49 [DEBUG] [aws-sdk-go] DEBUG: Request route53/CreateHostedZone Details:
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: ---[ REQUEST POST-SIGN ]-----------------------------
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: POST /2013-04-01/hostedzone HTTP/1.1
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Host: api.route53.cn
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: User-Agent: aws-sdk-go/1.19.42 (go1.12.5; darwin; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.12.1
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Content-Length: 311
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Authorization: AWS4-HMAC-SHA256 Credential=AKIA2SY65VLYF3DK2EHK/20190615/us-east-1/route53/aws4_request, SignedHeaders=content-length;host;x-amz-date, Signature=75b091758077c471fdda8d42b9a436955c0bbc59a23f5ca8b433c26c9ee43beb
> 2019-06-15T12:15:49.099+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: X-Amz-Date: 20190615T041549Z
> 2019-06-15T12:15:49.100+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Accept-Encoding: gzip
> 2019-06-15T12:15:49.100+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 
> 2019-06-15T12:15:49.100+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: <CreateHostedZoneRequest xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><HostedZoneConfig><Comment>try to create a pub zone with terraform in China region</Comment></HostedZoneConfig><Name>bobtest2.cn</Name><CallerReference>terraform-20190615041549099100000001</CallerReference></CreateHostedZoneRequest>
> 2019-06-15T12:15:49.100+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: -----------------------------------------------------
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/15 12:15:49 [DEBUG] [aws-sdk-go] DEBUG: Response route53/CreateHostedZone Details:
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: ---[ RESPONSE ]--------------------------------------
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: HTTP/1.1 403 Forbidden
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Connection: close
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Content-Length: 317
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Content-Type: text/xml
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: Date: Sat, 15 Jun 2019 04:15:48 GMT
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: X-Amzn-Requestid: 40d95499-8f24-11e9-b1d3-27c91871b3b6
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: -----------------------------------------------------
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/15 12:15:49 [DEBUG] [aws-sdk-go] <?xml version="1.0"?>
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: <ErrorResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><Error><Type>Sender</Type><Code>SignatureDoesNotMatch</Code><Message>Credential should be scoped to a valid region, not 'us-east-1'. </Message></Error><RequestId>40d95499-8f24-11e9-b1d3-27c91871b3b6</RequestId></ErrorResponse>
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/15 12:15:49 [DEBUG] [aws-sdk-go] DEBUG: Validate Response route53/CreateHostedZone failed, not retrying, error SignatureDoesNotMatch: Credential should be scoped to a valid region, not 'us-east-1'. 
> 2019-06-15T12:15:49.667+0800 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 	status code: 403, request id: 40d95499-8f24-11e9-b1d3-27c91871b3b6
> 2019/06/15 12:15:49 [DEBUG] aws_route53_zone.test: apply errored, but we're indicating that via the Error pointer rather than returning it: error creating Route53 Hosted Zone: SignatureDoesNotMatch: Credential should be scoped to a valid region, not 'us-east-1'. 
> 	status code: 403, request id: 40d95499-8f24-11e9-b1d3-27c91871b3b6
> 2019/06/15 12:15:49 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
> 2019/06/15 12:15:49 [ERROR] <root>: eval: *terraform.EvalSequence, err: error creating Route53 Hosted Zone: SignatureDoesNotMatch: Credential should be scoped to a valid region, not 'us-east-1'. 
> 	status code: 403, request id: 40d95499-8f24-11e9-b1d3-27c91871b3b6
> 2019/06/15 12:15:49 [TRACE] [walkApply] Exiting eval tree: aws_route53_zone.test
> 2019-06-15T12:15:49.691+0800 [DEBUG] plugin: plugin process exited: path=/Users/shengbo/ssh/terra/.terraform/plugins/darwin_amd64/terraform-provider-aws_v2.14.0_x4 pid=1801
> 2019-06-15T12:15:49.691+0800 [DEBUG] plugin: plugin exited

### Panic Output


### Expected Behavior
In China region, we have to specify the region as  "cn-northwest-1" for route53 service

### Actual Behavior

the region is hard code to "us-east-1"  in config.go;

`r53conn:                             route53.New(sess.Copy(&aws.Config{Region: aws.String("us-east-1"), Endpoint: aws.String(c.Endpoints["route53"])})),`

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids



### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

--- a/aws/config.go
+++ b/aws/config.go
@@ -481,6 +481,13 @@ func (c *Config) Client() (interface{}, error) {
 		globalAcceleratorConfig.Region = aws.String(endpoints.UsWest2RegionID)
 		route53Config.Region = aws.String(endpoints.UsEast1RegionID)
 		shieldConfig.Region = aws.String(endpoints.UsEast1RegionID)
+	case endpoints.AwsCnPartitionID:
+		// The AWS Go SDK is missing endpoint information for Route 53 in the AWS China partition.
+		// This can likely be removed in the future.
+		if aws.StringValue(route53Config.Endpoint) == "" {
+			route53Config.Endpoint = aws.String("https://api.route53.cn")
+		}
+		route53Config.Region = aws.String(endpoints.CnNorthwest1RegionID)
 	case endpoints.AwsUsGovPartitionID:
 		// The AWS Go SDK is missing endpoint information for Route 53 in the AWS GovCloud (US) partition.
 		// This can likely be removed in the future.

